### PR TITLE
Release 3.8.0 fixes for the related discussions

### PR DESCRIPTION
**Pull request by christyfernandes**

### CODE REVIEW

**Review by vinukumar-vs — COMMENTED** [review]
@christyfernandes Review this PR once.

File: projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts [hunk 1]
```diff
@@ -68,19 +71,22 @@ export class DiscussionDetailsComponent implements OnInit, OnDestroy {
...
         this.routeParams = params;
         this.slug = _.get(this.routeParams, 'slug');
         this.topicId = _.get(this.routeParams, 'topicId');
-        this.refreshPostData(this.currentActivePage);
+        this.cid = await this.refreshPostData(this.currentActivePage);
+        // this.getRealtedDiscussion(this.cid)
       });
     } else {
-      this.refreshPostData(this.currentActivePage);
+      this.cid = await this.refreshPostData(this.currentActivePage);
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts:85]: Why refreshPostData is returning the CategoryId?

File: projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts [hunk 2]
```diff
@@ -68,19 +71,22 @@ export class DiscussionDetailsComponent implements OnInit, OnDestroy {
...
+      // this.getRealtedDiscussion(this.cid)
     }
 
+
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts:89]: Why do we need below method paramsSubscription

Comment (vinukumar-vs) [reply projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts:89]: We are doing the same on ngOnit()

File: projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts [hunk 3]
```diff
@@ -103,6 +113,10 @@ export class DiscussionDetailsComponent implements OnInit, OnDestroy {
       })
   }
 
+  getRealtedDiscussion(catId, page?: any) {
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts:116]: This method is not used anywhere?

File: projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts [hunk 4]
```diff
@@ -115,35 +129,48 @@ export class DiscussionDetailsComponent implements OnInit, OnDestroy {
...
-          // this.setPagination();
-        },
-        (err: any) => {
-          // toast message
-          // this.openSnackbar(err.error.message.split('|')[1] || this.defaultError);
+      try {
+        return new Promise(async (resolve, reject) => {
+          this.discussionService.fetchTopicById(this.topicId, this.slug, page).subscribe(
+            (data: NSDiscussData.IDiscussionData) => {
+              this.data = data;
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts:138]: Duplication of the code in if else methods

File: projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts [hunk 5]
```diff
@@ -1,27 +1,61 @@
...
   host: { class: 'margin-left-l' },
 })
-export class RelatedDiscussionComponent implements OnInit {
-  @Input()
-  relatedDiscussions!: []
+export class RelatedDiscussionComponent implements OnInit, OnChanges {
+  @Input() catId: any
+  @Input() topicId: any
+  relatedDiscussions = []
+  categoryId!: number
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts:18]: Why are we using catID & CategoryId as two different private variables?

File: projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts [hunk 6]
```diff
@@ -1,27 +1,61 @@
...
+    private discussionService: DiscussionService,
   ) { }
 
   ngOnInit() {
+
+  }
+
+  ngOnChanges() {
+    if (this.catId) {
+      this.fetchSingleCategoryDetails(this.catId)
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts:34]: Why the method name is called as singleCategory?

File: projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts [hunk 7]
```diff
@@ -1,27 +1,61 @@
...
+    }
+  }
+
+  fetchSingleCategoryDetails(cid: number) {
+    this.fetchSingleCategoryLoader = true
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts:39]: Change the variable name

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.html [hunk 8]
```diff
@@ -53,12 +53,12 @@
                 <a href="javascript:void(0)" i18n-aria-label role="link" *ngIf="!post.upvoted" (click)="upvote(post);logTelemetry($event, post)"
                   aria-label="upvote the post" id="up-vote">
                   <!-- upward arrow -->
-                  <img src="./assets/discussion-ui/images/up-arrow.png" alt="up-arrow" width="14px" height="14px" class="icons">
+                  <img src="./assets/discussion-ui/images/up-arrow.png" width="14px" height="14px" class="icons">
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.html:56]: Why did you removed this "alt" property? Which is required on mouse-over to show the text. 
Get this property back for all the images

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts [hunk 9]
```diff
@@ -46,6 +46,9 @@ export class DiscussionDetailsComponent implements OnInit, OnDestroy {
   showEditTopicModal = false;
   editableTopicDetails: any;
   dropdownContent = true;
+  cid: any;
+  catId: any;
```
[the remark this answers is not in the document]
Comment (christyfernandes) [reply projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts]: Resolved

File: projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.html (hunk 8 above)
Comment (christyfernandes) [reply projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.html]: This was removed by mistake while resolving conflicts

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**